core/src/executor/update.rs

```diff
@@ -110,6 +117,36 @@ impl<'a, T: GStore> Update<'a, T> {
                     Ok::<_, Error>((id.as_ref(), value))
                 }
             })
+            .and_then(|(id, value)| async move {
+                for foreign_key in foreign_keys {
+                    let ForeignKey {
+                        referencing_column_name,
+                        referenced_table_name,
+                        referenced_column_name,
+                        ..
+                    } = foreign_key;
+                    if referencing_column_name != id || value == Value::Null {
+                        return Ok((id, value));
+                    }
```

Comment:
1. `value == Value::Null` check can be done before the foreign key for loop?
2. we may need.. `continue` instead of `return`?